cray_product_catalog/catalog_update.py

```diff
@@ -155,7 +170,7 @@ def update_config_map(data: dict, name, namespace):
     api_instance = client.CoreV1Api(k8sclient)
     attempt = 0
 
-    while True:
+    while attempt < retries:
```

Comment:
This is overloading the `retries` variable. The `retries` is used in the `Retry` object created above and used in the K8s client, which is different from the number of attempts in this loop. It used to attempt infinitely. I'm not sure if that's necessarily a good thing, but you might still want to be careful about changing this logic and using the `retries` variable in two different ways.

Reply:
Can you please explain why this change is being made?

Reply:
If read_namespaced_config_map call fails with err.status as ERR_NOT_FOUND or create_namespaced_config_map call fails then it will go into infinite loop as we have continue at line 199. This is one of the reason we added the condition while attempt < retries. This will help in coming out of the loop.

Can we know the reason behind designing this as infinite loop?

Reply:
We also see that attempt is initialized to 0 initially and later incremented. But never used to exit the loop. Is this variable required?

Reply:
We decided that it is okay to use the `retries` variable in this way.